### PR TITLE
Improved documentation

### PR DESCRIPTION
**Pull request by int0x81**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description
This pull request aims to improve the projects documentation
- Fixed a typo in debug.md
- Created a reference documentation for the scaffolding process of new piral instances / pilets (Please review carefully)
- Setup a skeleton for the VS Code documentation
- Improved the gen-types.js to also work properly on Windows OS
- Updated some generated docs

### CODE REVIEW

File: docs/tools/vscode-piral.md [hunk 1]
```diff
@@ -4,4 +4,12 @@ title: Piral VSCode
...
-(tbd)
+Piral comes with a VS Code extension that's available at the official [marketplace](https://marketplace.visualstudio.com/items?itemName=smapiot.vscode-piral)
+
+### Features
+- Lets you execute all relevant piral-cli commands right from VS Code
+- Provides you with information about your current piral workspace
+
+
+
+*Note: This plugin is currently beeing developed*
```
Comment (FlorianRappl) [inline docs/tools/vscode-piral.md:15]: The note can be done via an special box:

```md
::: warning: Under Development
The plugin is currently being developed.
:::
```

File: docs/reference/scaffolding.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Scaffolding
```
Comment (FlorianRappl) [inline docs/reference/scaffolding.md:1]: Link this markdown in the `docs/reference/README.md`.

File: docs/reference/scaffolding.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+This section will help you with the process of creating fresh piral and pilet instances.
+
+
+### Get started
+The simplest way to scaffold a piral project is to use the piral-cli by installing:
+
+`npm i -g piral-cli`
```
[the remark this answers is not in the document]
Comment (int0x81) [reply docs/reference/scaffolding.md]: adapted throughout the whole document

File: docs/reference/scaffolding.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Scaffolding
+This section will help you with the process of creating fresh piral and pilet instances.
```
[the remark this answers is not in the document]
Comment (int0x81) [reply docs/reference/scaffolding.md]: adapted throughout the whole document

File: docs/reference/scaffolding.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+### Selecting a bundler
+
+When scaffolding a piral instance or pilets you can also specify the bundler:
+
+`piral new myPiralAppShell --bundler parcel`
+
+Possible choices are `"none"`, `"parcel"` or `"webpack"`. The bundler can also be changed after the scaffolding process at any time.
+
+### Related errors
```
[the remark this answers is not in the document]
Comment (int0x81) [reply docs/reference/scaffolding.md]: Removed the 'Related errors' section

File: docs/messages/0041.md [hunk 6]
```diff
@@ -4,42 +4,12 @@ The section "piral" in the "package.json" file is missing.
...
the

```sh
pilet new
```

command.

Specifically, the package.json needs to contain a special key called `piral`, which
contains an object with additional fields.
```
Comment (FlorianRappl) [inline docs/messages/0041.md:7]: Ah apparently it did have an impact on git.

Can you please revert this and set your global git attributes to only use `\n` and not `\r\n`? The additional `\r` is quite annoying and should not go into the repo. Thanks!

File: docs/reference/scaffolding.md (hunk 2 above)
Comment (int0x81) [reply docs/reference/scaffolding.md:1]: Is now beeing linked in the README.md

File: docs/reference/scaffolding.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Scaffolding
+This section will help you with the process of creating fresh piral and pilet instances.
+
```
[the remark this answers is not in the document]
Comment (int0x81) [reply docs/reference/scaffolding.md:2]: adapted

File: docs/messages/0041.md (hunk 6 above)
Comment (int0x81) [reply docs/messages/0041.md:7]: Yep I will change it. Sadly these changes are hard to see in vscode working trees :\

Comment (FlorianRappl) [reply docs/messages/0041.md:7]: Agreed.

Comment (int0x81) [reply docs/messages/0041.md]: I pushed the updated generated docs (11 Files). Now with the correct line endings

File: src/pages/docs/src/codegen/gen-types.js [hunk 8]
```diff
@@ -12,6 +12,8 @@ module.exports = function() {
   const imports = files.map(file => {
     const body = readFileSync(file, 'utf8');
     const name = file
+      .split("\\")
```
Comment (FlorianRappl) [inline src/pages/docs/src/codegen/gen-types.js:15]: Prettify (single quotes).

File: docs/reference/scaffolding.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+### Logging
+
+If you want more information about the scaffolding process you can simply increase the log level (max 5):
+
+```sh
+piral new myPiralAppShell --log-level 5
+```
+
+### Selecting a bundler
```
Comment (FlorianRappl) [inline docs/reference/scaffolding.md:45]: Use title case "Selecting a Bundler"

File: docs/reference/scaffolding.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
+# Scaffolding
+
+This section will help you with the process of creating fresh piral and pilet instances.
+
+### Get started
```
Comment (FlorianRappl) [inline docs/reference/scaffolding.md:5]: Use title case "Get Started"

File: src/pages/docs/src/codegen/gen-types.js (hunk 8 above)
Comment (int0x81) [reply src/pages/docs/src/codegen/gen-types.js]: adapted

File: docs/reference/scaffolding.md (hunk 9 above)
Comment (int0x81) [reply docs/reference/scaffolding.md]: adapted

File: docs/reference/scaffolding.md (hunk 10 above)
Comment (int0x81) [reply docs/reference/scaffolding.md]: adapted

**Pull request merged**